Terminal.Gui/ConsoleDrivers/ConsoleKeyMapping.cs

```diff
@@ -3,825 +3,1626 @@
+
+		/// <summary>
+		/// Attn key
+		/// </summary>
+		OEM_ATTN = 0xF0,
+
+		/// <summary>
+		/// Finish key
+		/// </summary>
+		OEM_FINISH = 0xF1,
+
+		/// <summary>
+		/// Copy key
+		/// </summary>
+		OEM_COPY = 0xF2,
+
+		/// <summary>
+		/// Auto key
+		/// </summary>
+		OEM_AUTO = 0xF3,
+
+		/// <summary>
+		/// Enlw key
+		/// </summary>
+		OEM_ENLW = 0xF4,
+
+		/// <summary>
+		/// BackTab key
+		/// </summary>
+		OEM_BACKTAB = 0xF5,
+
+		/// <summary>
+		/// Attn key
+		/// </summary>
+		ATTN = 0xF6,
+
+		/// <summary>
+		/// CrSel key
+		/// </summary>
+		CRSEL = 0xF7,
+
+		/// <summary>
+		/// ExSel key
+		/// </summary>
+		EXSEL = 0xF8,
+
+		/// <summary>
+		/// Erase EOF key
+		/// </summary>
+		EREOF = 0xF9,
+
+		/// <summary>
+		/// Play key
+		/// </summary>
+		PLAY = 0xFA,
+
+		/// <summary>
+		/// Zoom key
+		/// </summary>
+		ZOOM = 0xFB,
+
+		/// <summary>
+		/// Reserved
+		/// </summary>
+		NONAME = 0xFC,
+
+		/// <summary>
+		/// PA1 key
+		/// </summary>
+		PA1 = 0xFD,
+
+		/// <summary>
+		/// Clear key
+		/// </summary>
+		OEM_CLEAR = 0xFE
+	}
+
+
+	static HashSet<ScanCodeMapping> _scanCodes = new HashSet<ScanCodeMapping> {
+		new (1, VK.ESCAPE, 0, '\u001B'), // Escape
+		new (1, VK.ESCAPE, ConsoleModifiers.Shift, '\u001B'),
+		new (2, (VK)'1', 0, '1'), // D1
+		new (2, (VK)'1', ConsoleModifiers.Shift, '!'),
+		new (3, (VK)'2', 0, '2'), // D2
+		new (3, (VK)'2', ConsoleModifiers.Shift, '\"'), // BUGBUG: This is true for Portugese keyboard, but not ENG (@) or DEU (")
+		new (3, (VK)'2', ConsoleModifiers.Alt | ConsoleModifiers.Control, '@'),
+		new (4, (VK)'3', 0, '3'), // D3
+		new (4, (VK)'3', ConsoleModifiers.Shift, '#'),
+		new (4, (VK)'3', ConsoleModifiers.Alt | ConsoleModifiers.Control, '£'),
+		new (5, (VK)'4', 0, '4'), // D4
+		new (5, (VK)'4', ConsoleModifiers.Shift, '$'),
+		new (5, (VK)'4', ConsoleModifiers.Alt | ConsoleModifiers.Control, '§'),
+		new (6, (VK)'5', 0, '5'), // D5
+		new (6, (VK)'5', ConsoleModifiers.Shift, '%'),
+		new (6, (VK)'5', ConsoleModifiers.Alt | ConsoleModifiers.Control, '€'),
+		new (7, (VK)'6', 0, '6'), // D6
+		new (7, (VK)'6', ConsoleModifiers.Shift, '&'),
+		new (8, (VK)'7', 0, '7'), // D7
+		new (8, (VK)'7', ConsoleModifiers.Shift, '/'),
+		new (8, (VK)'7', ConsoleModifiers.Alt | ConsoleModifiers.Control, '{'),
+		new (9, (VK)'8', 0, '8'), // D8
+		new (9, (VK)'8', ConsoleModifiers.Shift, '('),
+		new (9, (VK)'8', ConsoleModifiers.Alt | ConsoleModifiers.Control, '['),
+		new (10, (VK)'9', 0, '9'), // D9
+		new (10, (VK)'9', ConsoleModifiers.Shift, ')'),
+		new (10, (VK)'9', ConsoleModifiers.Alt | ConsoleModifiers.Control, ']'),
+		new (11, (VK)'0', 0, '0'), // D0
+		new (11, (VK)'0', ConsoleModifiers.Shift, '='),
+		new (11, (VK)'0', ConsoleModifiers.Alt | ConsoleModifiers.Control, '}'),
+		new (12, VK.OEM_4, 0, '\''), // Oem4
+		new (12, VK.OEM_4, ConsoleModifiers.Shift, '?'),
+		new (13, VK.OEM_6, 0, '+'), // Oem6
+		new (13, VK.OEM_6, ConsoleModifiers.Shift, '*'),
```

Comment:
Not sure what's going on here, but if I enable the "POR" keyboard and press the Shift-Oem6 key (labeled `=`/`+` on the ENG keyboard) I see this:

```
event: [KeyEventRecord(down,1,OEM_6,13,»,ShiftPressed)] [ConsoleKeyInfoEx(Key: 221 (Oem6) | Shift, KeyChar: » (187))]
event: [KeyEventRecord(up,1,OEM_6,13,»,ShiftPressed)] [ConsoleKeyInfoEx(Key: 221 (Oem6) | Shift, KeyChar: » (187))]
```

> I can't send to the second TextView on the Simulator |!"#$%&/()=?»>;:_.

At this point, I do not care about the simulator. It serves no purpose right now other than to be a distraction. It doesn't actually simulate anything important. I am focused on fixing keyboard handling in a keyboard layout-independent way. That has nothing to with `VK_PACKET` nor the simulator.

Reply:
These are the values. You changed many of these and you must check it it's all right. It appears as `OEM_6` because you configured with that value on the scan code equal to 13.

![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/62d87999-643a-4f67-bf30-922f34c72077)

![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/fc44d13b-dd40-4200-9ace-27dee917610c)

Reply:
> Not sure what's going on here, but if I enable the "POR" keyboard and press the Shift-Oem6 key (labeled =/+ on the ENG keyboard) I see this:

Mine is labeled `»`/`«` on POR keyboard 
![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/1179a557-db9f-4a50-884e-d6d4372417cb).

Reply:
I think this is going to be a lot of work, like trying to resurrect a dead person. If you think it's the correct behavior because on the POR layout of your keyboard is right, then we cannot have an agreement here because on mine it isn't correct.

Reply:
If I use the ENG layout I get the `OemPlus` which is the correct behavior. So something is wrong.

![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/807a6898-42bb-4eeb-8098-eca236c398bf)

![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/ca04a60e-0cce-4ec9-ac20-9f08ce86b68b)

![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/0254ffc4-22d2-4788-9f1e-0d28a75fc31b)

![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/43636337-6cac-45c4-8228-c31320e9f83f)

Reply:
I just pushed a quick change that will print the keyboard layout name on each keypress.

Note that `GetKeybaordLayoutName` uses the current threadID and thus will only report the correct name if you did NOT change the keyboard layout after the debugging session was started.

Please 

- Set your keyboard layout to POR
- debug UICatalog
- Press the `»`/`«` key

Send what you see in the debug output. I see:

```
event: KBD: 00000816 [KeyEventRecord(down,1,OEM_6,13,«,0)] [ConsoleKeyInfoEx(Key: 221 (Oem6), KeyChar: « (171))]
event: KBD: 00000816 [KeyEventRecord(up,1,OEM_6,13,«,0)] [ConsoleKeyInfoEx(Key: 221 (Oem6), KeyChar: « (171))]
```

If I re-run the above, but first set my keyboard layout to ENG I see:

```
event: KBD: 00000409 [KeyEventRecord(down,1,OEM_PLUS,13,=,0)] [ConsoleKeyInfoEx(Key: 187 (OemPlus), KeyChar: = (61))]
event: KBD: 00000409 [KeyEventRecord(up,1,OEM_PLUS,13,=,0)] [ConsoleKeyInfoEx(Key: 187 (OemPlus), KeyChar: = (61))]
```

Both of these are correct.

Reply:
It's the same as your.

```
event: KBD: 00000816 [KeyEventRecord(down,1,OEM_6,13,«,0)] [ConsoleKeyInfoEx(Key: 221 (Oem6), KeyChar: « (171))]
event: KBD: 00000816 [KeyEventRecord(up,1,OEM_6,13,«,0)] [ConsoleKeyInfoEx(Key: 221 (Oem6), KeyChar: « (171))]
```

```
event: KBD: 00000409 [KeyEventRecord(down,1,OEM_PLUS,13,=,0)] [ConsoleKeyInfoEx(Key: 187 (OemPlus), KeyChar: = (61))]
event: KBD: 00000409 [KeyEventRecord(up,1,OEM_PLUS,13,=,0)] [ConsoleKeyInfoEx(Key: 187 (OemPlus), KeyChar: = (61))]
```

Reply:
Good. So it's all working correctly now. Yay!